portfolio/src/main/webapp/script.js

```diff
@@ -12,12 +12,115 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+var context;
+var canvas;
+var myBubbleArray;
+var numBubs = 20;
+var animationID = null;
+// Resizes bubbles when page size is changed
+window.addEventListener('resize', init)
+
+function startChat() {
+  grabComments();
+  init();
+  grabUser();
+}
+
+class Bubble {
+  constructor(color) {
+    this.speed_factor = 3;
+    if (canvas.width < 900) {
+      this.speed_factor = 1.75;
+    }
+    this.color = color;
+    this.size = Math.random() * canvas.width;
```

Comment:
The constructor should take a canvas argument instead of relying on a global variable. It should also store the canvas in an instance variable.